node/node.go

```diff
@@ -960,6 +960,10 @@ func (n *Node) GetAccountWithKeys(address string, options api.AccountQueryOption
 		return api.AccountResponse{}, api.BlockInfo{}, err
 	}
 
+	if accInfo.account == nil || accInfo.account.DataTrie() == nil {
```

Comment:
is this method only used when options.WithKeys is set? 

if yes then all good, otherwise maybe you need to move the check for DataTrie in the code block starting at line 968

Reply:
moved inside the if on line 968